doc: cleanup

justinmk opened the pull request:
- Remove `nvim-` prefix from most tags, except those in `vim_diff.txt`. This prefix is completely unnecessary.
- Reorganize and merge some sections.
- Edit for brevity and clarity.
- Beginning of neovim-specific `develop.txt`.

justinmk committed:
doc: Move spell design discussion.

It is worth preserving, but let it live in spell.txt so we can dedicate
develop.txt to nvim-specific discussion.

justinmk committed:
doc: msgpack_rpc: De-emphasize the transport protocol.

Tighten our jargon, avoid mentioning "msgpack" everywhere we mention the
RPC API. Prefer "RPC API" when speaking about the remote API.

Though it's conceivable that we may one day support some protocol other
than msgpack, that isn't relevant to most of our discussion of the API,
including this document.

The file name msgpack_rpc.txt is preserved to avoid totally breaking
URLs.

justinmk committed:
doc: remove "nvim-" qualfier from tags

It is almost never necessary to qualify documentation as "Nvim
specific".

The main exception to this is vim_diff.txt.

justinmk committed:
doc: Remove gui_w32.txt

It's no longer relevant, except for 2 sections which are tracked as issues #4927
and #4928.

justinmk committed:
doc/clipboard: cleanup

justinmk committed:
doc/terminal: cleanup

justinmk committed:
doc: cleanup

justinmk committed:
doc/provider: cleanup

- Move design/impl discussion to develop.txt
- Move clipboard notes to provider.txt

justinmk committed:
doc/python: cleanup

- Move info to providers.txt
- Remove "nvim-" prefix.
- Brevity, clarity, ...

justinmk committed:
doc: uppercase RPC

justinmk commented on runtime/doc/provider.txt:
```diff
@@ -9,7 +9,86 @@ Providers	 						    *provider*
+    $ sudo pip2 install --upgrade neovim
+<
+  or for the current user: >
+    $ pip2 install --user --upgrade neovim
```
Always mention `--upgrade`, there's no reason not to.

justinmk commented on runtime/doc/provider.txt:
```diff
@@ -9,7 +9,86 @@ Providers	 						    *provider*
 
 ==============================================================================
-Clipboard integration 						   *clipboard*
+Python integration		    	      *provider-python*
```
Prefix providers with `provider-`.

oni-link commented on runtime/doc/nvim.txt:
```diff
@@ -14,7 +14,7 @@ see |help.txt|.
 3. Msgpack-RPC			 |msgpack-rpc|
 4. Job control			 |job-control|
-5. Python plugins		 |nvim-python|
+5. Python plugins		 |python-provider|
```
`provider-python`?

oni-link commented on runtime/doc/nvim_from_vim.txt:
```diff
@@ -16,8 +16,8 @@ it:
 <
-See |nvim-intro|, especially |nvim-python| and |clipboard|, for additional
-software you might need to install to use all of Nvim's features.
+See |python-provider| and |provider-clipboard| for additional software you
```
`provider-python`?

rumpelsepp commented on runtime/autoload/provider/clipboard.vim:
```diff
@@ -58,7 +58,7 @@ elseif executable('doitclient')
   let s:paste['*'] = s:paste['+']
 else
-  echom 'clipboard: No clipboard tool available. See :help nvim-clipboard'
+  echom 'clipboard: No clipboard tool available. :help clipboard'
```
I would leave the 'See'; IMO it sounds strange without it.

KillTheMule commented:
Since you're at it, you could add the statusline alignment improvement to vim-diff.txt, and the `:CheckHealth` command to nvim_python.txt. We should advertise new features more :)

jamessan commented on runtime/doc/gui_w32.txt:
```diff
@@ -1,262 +1,14 @@
+Vim for Win32
 
 ==============================================================================
 4. Shell Commands					*gui-shell-win32*
```
Should the section numbers be removed since you got rid of the table of contents?

justinmk replied:
@jamessan I created proper issues and will delete gui_w32.txt entirely.

justinmk commented on runtime/doc/msgpack_rpc.txt:
```diff
@@ -1,97 +1,88 @@
->
-    nvim --api-info | python -c 'import msgpack, sys, yaml; print yaml.dump(msgpack.unpackb(sys.stdin.read()))' > api.yaml
+To get a human-readable list of API functions, try this one-liner: >
+    nvim --api-info | nvim --headless +'put =msgpackparse([getline(1)])|g/\V\^{\._TYPE\.: [], \._VAL\.: [\.\w\+_\w\+\.]}\$/p' +'q!' -
```
Less terrible one-liner from @mhinz:

    :echomsg string(map(msgpackparse(systemlist('nvim --api-info'))[0].functions, 'v:val.name._VAL[0]'))

bfredl replied:
why not allow `string(map(v:api_info.functions, 'v:val.name'))` ?

justinmk committed:
doc: consolidate nvim.txt

justinmk committed:
build: Purge docs before rebuilding.

If a help file is renamed, stale help files in the build workspace will
cause duplicate tags (which causes the build to fail). To avoid this,
always delete build/runtime/doc/ before building helptags.
